xml.go

```diff
@@ -456,7 +464,14 @@ func xmlToMapParser(skey string, a []xml.Attr, p *xml.Decoder, r bool) (map[stri
 			return n, nil
 		case xml.CharData:
 			// clean up possible noise
-			tt := string(t.(xml.CharData))
+			sb := strings.Builder{}
+			sb.WriteString("\t\r\b\n")
+			fmt.Printf("Trim white space? %v\n", disableTrimWhiteSpace)
+			if !disableTrimWhiteSpace {
+				sb.WriteRune(' ')
+			}
+
+			tt := strings.Trim(string(t.(xml.CharData)), sb.String())
 			if len(tt) > 0 {
 				if len(na) > 0 || decodeSimpleValuesAsMap {
 					na["#text"] = cast(tt, r, "#text")
```

Comment:
This is adding a lot of processing overhead, since case xml.CharData is the common 'case' during decoding.  Need something like - 

```
var trimRunes string = "\t\r\b\n "
var disableTrimWhiteSpace bool

// ... document ...
func DisableTrimWhiteSpace(b ...bool) {
    if len(b) == 0 {
      disableTrimWhiteSpace = !disableTrimWhiteSpace
   } else {
      disableTrimWhiteSpace = b[0]
   }

   if disableTrimWhiteSpace {
      trimRunes = "\t\r\b\n"
   } else {
      trimRunes = "\t\r\b\n "
   }
}

...
                                    case xml.CharData:
                                       // clean up possible noise
                                       tt := stringsTrim(string(t.(xml.CharData))l, trimRunes)
                                       ...
```

Reply:
```
func DisableTrimWhiteSpace(b ...bool) {
    if len(b) == 0 {
      disableTrimWhiteSpace = !disableTrimWhiteSpace
```

Hmmm, not too sure if this should be a toggle. It would be hard to tell what the final result if `DisableTrimWhiteSpace` is called several times. Given the name this should probably always set it do false unless a boolean is provided.

But yeah, I agree that it is a good idea to create the rune when setting the configuration to prevent rebuilding the trimRunes. I'll make the change.